javaslang-test/src/main/java/javaslang/test/Gen.java

```diff
@@ -277,6 +310,17 @@
         return choose(1, size).flatMap(n -> new Frequency().gen(n, stream));
     }
 
+  /**
+   * Intersperse values from this generator instance with those of another.
+   *
+   * @param other another T generator to accept values from.
+   * @return A new T generator
+   */
+    default Gen<T> intersperse(Gen<T> other) {
+        AtomicReference<Tuple2<Gen<T>, Gen<T>>> interspersedReference = new AtomicReference<>(Tuple.of(other, this));
+        return random -> interspersedReference.updateAndGet(generators -> Tuple.of(generators._2, generators._1))._1.apply(random);
```

Comment:
I also don't overly like creating new `Tuple` instances on every iteration, do we have any notion of a "rotate" or "shift" in javaslang that could maintain the underlying structure?